syllabus/examples/training_scripts/cleanrl_procgen_robustplr.py

```diff
@@ -0,0 +1,513 @@
+    stddev_returns = np.std(eval_episode_rewards)
+    env_min, env_max = PROCGEN_RETURN_BOUNDS[args.env_id]
+    normalized_mean_returns = (mean_returns - env_min) / (env_max - env_min)
+    policy.train()
+    return mean_returns, stddev_returns, normalized_mean_returns
+
+
+def make_value_fn():
+    def get_value(obs):
+        obs = np.array(obs)
+        with torch.no_grad():
+            return agent.get_value(torch.Tensor(obs).to(device))
+    return get_value
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    run_name = f"{args.env_id}__{args.exp_name}__{args.seed}__{int(time.time())}"
+    if args.track:
+        import wandb
+
+        wandb.init(
+            project=args.wandb_project_name,
+            entity=args.wandb_entity,
+            sync_tensorboard=True,
+            config=vars(args),
+            name=run_name,
+            monitor_gym=True,
+            save_code=True,
+            dir=args.logging_dir
+        )
+        wandb.run.log_code(os.path.join(args.logging_dir, "/syllabus/examples"))
+
+    writer = SummaryWriter(os.path.join(args.logging_dir, "./runs/{run_name}"))
+    writer.add_text(
+        "hyperparameters",
+        "|param|value|\n|-|-|\n%s" % ("\n".join([f"|{key}|{value}|" for key, value in vars(args).items()])),
+    )
+
+    # TRY NOT TO MODIFY: seeding
+    random.seed(args.seed)
+    np.random.seed(args.seed)
+    torch.manual_seed(args.seed)
+    torch.backends.cudnn.deterministic = args.torch_deterministic
+
+    device = torch.device("cuda" if torch.cuda.is_available() and args.cuda else "cpu")
+    print("Device:", device)
+
+    # Curriculum setup
+    task_queue = update_queue = lock = None
+    if args.curriculum:
+        sample_env = openai_gym.make(f"procgen-{args.env_id}-v0")
+        sample_env = GymV21CompatibilityV0(env=sample_env)
+        sample_env = ProcgenTaskWrapper(sample_env, args.env_id, seed=args.seed)
+
+        # Intialize Curriculum Method
+        if args.curriculum_method == "plr":
+            print("Using prioritized level replay.")
+            curriculum = PrioritizedLevelReplay(
+                task_space=sample_env.task_space,
+                observation_space=sample_env.observation_space,
+                action_space=sample_env.action_space,
+                eval_envs=sample_env.eval_envs,
+                num_steps=args.num_steps,
+                num_processes=args.num_envs,
+                gamma=args.gamma,
+                gae_lambda=args.gae_lambda,
+                task_sampler_kwargs_dict={"strategy": "value_l1"},
+                get_value=make_value_fn(),
+            )
+        elif args.curriculum_method == "dr":
+            print("Using domain randomization.")
+            curriculum = DomainRandomization(sample_env.task_space)
+        elif args.curriculum_method == "lp":
+            print("Using learning progress.")
+            curriculum = LearningProgressCurriculum(sample_env.task_space)
```

Comment:
Instead of adding a whole new file, just add another curriculum method option here in cleanrl_procgen_plr.py